test/PowerPod.js

```diff
@@ -1,60 +1,154 @@
+        };
+
+        const depositAndDelegateTo = async (st1inch, delegation, from, to, amount, duration = commonLockDuration) => {
+            await st1inch.connect(from).deposit(amount, duration);
+            await st1inch.connect(from).addPod(delegation.address);
+            await delegation.connect(from).delegate(to);
+        };
 
-        const St1inch = await ethers.getContractFactory('St1inch');
-        const st1inch = await St1inch.deploy(oneInch.address, expBase, addr.address);
-        await st1inch.deployed();
+        const accounts = await ethers.getSigners();
+        const [owner, alice] = accounts;
+
+        const oneInch = await deployContract('ERC20PermitMock', ['1inch', '1inch', owner.address, ether('200')]);
+        await oneInch.transfer(alice.address, ether('100'));
+
+        const st1inch = await deployContract('St1inch', [oneInch.address, expBase, owner.address]);
         await oneInch.approve(st1inch.address, ether('100'));
+        await oneInch.connect(alice).approve(st1inch.address, ether('100'));
 
-        const PowerPod = await ethers.getContractFactory('PowerPod');
-        const delegation = await PowerPod.deploy('PowerPod', 'PP', st1inch.address);
-        await delegation.deployed();
-        await stakeAndRegisterInDelegation(st1inch, delegation, addr1, amountToStake, 0);
+        const delegation = await deployContract('PowerPod', ['PowerPod', 'PP', st1inch.address]);
 
-        return { st1inch, delegation };
-    }
+        const whitelistRegistry = await deployContract('WhitelistRegistry', [delegation.address, BALANCE_THRESHOLD]);
+        // fill all whitelist into WhitelistRegistry
+        for (let i = 0; i < MAX_WHITELISTED; ++i) {
+            const userIndex = i + 2;
+            const user = accounts[userIndex];
+            await stakeAndRegisterInDelegation(st1inch, delegation, user, ether('2') * BigInt(i + 1), userIndex);
+            await whitelistRegistry.connect(user).register();
+        }
+        await stakeAndRegisterInDelegation(st1inch, delegation, owner, ether('1'), 0);
 
-    before(async function () {
-        accounts = await ethers.getSigners();
-        addr = accounts[0];
-        addr1 = accounts[1];
-    });
+        return {
+            contracts: { st1inch, delegation, whitelistRegistry },
+            accounts: { owner, alice, accounts },
+            functions: { depositAndDelegateTo },
+            other: { commonLockDuration },
+        };
+    }
 
     describe('Should calculate voting power', function () {
         it('for account with 0 balance', async function () {
-            const { delegation } = await loadFixture(initContracts);
-            expect(await delegation.votingPowerOf(addr.address)).to.equal(0);
+            const { contracts: { delegation }, accounts: { accounts } } = await loadFixture(initContracts);
+            expect(await delegation.votingPowerOf(accounts[9].address)).to.equal(0);
```

Comment:
Why `accounts[9]`?

Reply:
Because we need clear address without some balances. We can use anything address after fifth.

Reply:
renamed